aea/mail/base.py

```diff
@@ -45,17 +46,61 @@ class Empty(Exception):
     """Exception for when the inbox is empty."""
 
 
+class URI:
+    """URI following RFC3986."""
+
+    def __init__(self, uri_raw: str):
+        """Initialize the URI."""
+        self.uri_raw = uri_raw
+        parsed = urlparse(uri_raw)
```

Comment:
can this raise an exception? If yes, I think would be good to put in the docstring.